packages/indexer-agent/src/network.ts

```diff
@@ -468,11 +470,113 @@ export class Network {
     )
   }
 
+  async subgraphs(ids: string[]): Promise<Subgraph[]> {
+    if (ids.length == 0) {
+      return []
+    }
+    let subgraphs: Subgraph[] = []
+    const queryProgress = {
+      lastId: '',
+      first: 20,
+      fetched: 0,
+      exhausted: false,
+      retriesRemaining: 10,
+    }
+    this.logger.info(`Query subgraphs in batches of ${queryProgress.first}`)
+
+    while (!queryProgress.exhausted) {
+      this.logger.debug(`Query subgraphs by id`, {
+        queryProgress: queryProgress,
+        subgraphIds: ids,
+      })
+      try {
+        const result = await this.networkSubgraph.query(
+          gql`
+            query subgraphs(
+              $first: Int!
+              $lastId: String!
+              $subgraphs: [String!]!
+            ) {
+              subgraphs(
+                where: { id_gt: $lastId, id_in: $subgraphs }
+                orderBy: id
+                orderDirection: asc
+                first: $first
+              ) {
+                id
+                versionCount
+                versions {
+                  version
+                  createdAt
+                  subgraphDeployment {
+                    id
+                  }
+                }
+              }
+            }
+          `,
+          {
+            first: queryProgress.first,
+            lastId: queryProgress.lastId,
+            subgraphs: ids,
+          },
+        )
+
+        if (result.error) {
+          throw result.error
+        }
+
+        // Convert return object to Subgraph interface
+        // eslint-disable-next-line @typescript-eslint/no-explicit-any
+        const results = result.data.subgraphs.map((subgraph: any) => {
+          subgraph.versions = subgraph.versions.map(
+            // eslint-disable-next-line @typescript-eslint/no-explicit-any
+            (versionItem: any) => {
+              return {
+                version: versionItem.version,
+                createdAt: versionItem.createdAt,
+                deployment: new SubgraphDeploymentID(
+                  versionItem.subgraphDeployment.id,
+                ),
+              } as SubgraphVersion
+            },
+          )
+          return subgraph
+        })
+
+        // In the case of a fresh graph network there will be no published subgraphs, handle gracefully
+        if (results.length == 0) {
+          return []
```

Comment:
I remember there was a fix relating to this since _indexer erroneously close allocations_? was it handled somewhere else?

Reply:
Good spot, I need to rebase this branch onto latest `main`.